pyqtorch/primitive.py

```diff
@@ -147,7 +147,9 @@ def __init__(self, control: int, target: int):
 
 
 class CSWAP(Primitive):
-    def __init__(self, control: int | tuple[int, ...], target: int):
+    def __init__(self, control: tuple[int, ...], target: int):
```

Comment:
the target is always a tuples since SWAP acts on two qubits. the control however can be int or tuple[int,..]

Reply:
So this version is correct right @jpmoutinho: 
```python3
class CSWAP(Primitive):
    def __init__(self, control: int | tuple[int, ...], target: tuple[int, ...]):
        if not isinstance(target, tuple) or len(target) == 2:
            raise ValueError("Target qubits must be a tuple with two qubits")
        super().__init__(OPERATIONS_DICT["CSWAP"], target)
        self.control = (control,) if isinstance(control, int) else control
        self.target = target
        self.qubit_support = self.control + (target,)
```